patches/tModLoader/Terraria/ModLoader/ModNet.cs

```diff
@@ -268,21 +271,115 @@ internal static bool SyncClientMods(BinaryReader reader, out bool needsReload)
 			return false;
 		}
 
-		// ready to connect, apply configs. Config manager will apply the configs on reload automatically
-		if (!needsReload) {
+		// Even if needsReload, check configs anyway so user is more informed.
+		HashSet<string> modsWithConfigReload = new HashSet<string>();
+		foreach (NetConfig pendingConfig in pendingConfigs) {
+			if (modsWithConfigReload.Contains(pendingConfig.modname) || !ModLoader.HasMod(pendingConfig.modname))
+				continue;
+
+			ModConfig currentConfigClone = ConfigManager.GeneratePopulatedClone(ConfigManager.GetConfig(pendingConfig));
+			JsonConvert.PopulateObject(pendingConfig.json, currentConfigClone, ConfigManager.serializerSettingsCompact);
+			Mod mod = ModLoader.GetMod(pendingConfig.modname);
+			ModConfig loadTimeConfig = ConfigManager.GetLoadTimeConfig(mod, pendingConfig.configname);
+
+			if (loadTimeConfig.NeedsReload(currentConfigClone)) {
+				needsReload = true;
+				reloadRequiredExplanationEntries.Add(new ReloadRequiredExplanation(5, mod.Name, modFiles.FirstOrDefault(mod => mod.Name == pendingConfig.modname), Language.GetTextValue("tModLoader.ReloadRequiredExplanationConfigChanged", "DDA0DD")));
+				modsWithConfigReload.Add(pendingConfig.configname);
```

Comment:
Can we factor this into something like `ConfigManager.AnyModNeedsReloadCheckOnly`? Can be a method here, or in `ConfigManager`. Something like `(List<NetConfig> pendingConfigs, out List<Mod> modsWithChangedConfigs)`. You can still keep the debug log message in the method. If we wanted to report the config(s) we could change it to `out List<ModConfig>` instead at a later time